### PR TITLE
fix: update cosmos methods to match exodus provider methods

### PR DESCRIPTION
**Pull request by mzndako**
# Summary
Update cosmos methods to match corresponding methods in the exodus provider

### CODE REVIEW

**Review by timlanahan — APPROVED** [review]
tACK

tested connection in the preview link but let's wait for @bulgakovk to review also

File: wallets/exodus-extension/src/extension/client.ts
Comment (bulgakovk) [inline wallets/exodus-extension/src/extension/client.ts]: Just curious if a dapp expects that `getOfflineSigner` initiates a connection flow. Asking because `getAccounts` alternatively can be implemented like:
```
getAccounts: async (): Promise<AccountData[]> => this.getAccount(chainId)
```

**Review by bulgakovk — APPROVED** [review]
utACK (checked that all types comply with the Exodus provider interface)

Note: once this PR is merged, we'll need to update our fork and include a wallet logo as per https://docs.cosmoskit.com/integrating-wallets/adding-new-wallets#about-wallet-logos

File: wallets/exodus-extension/src/extension/client.ts
Comment (mzndako) [reply wallets/exodus-extension/src/extension/client.ts]: not sure but the reason for calling the method (`getAccount`) there is to make sure `this.account` has been set before. Will update it to use your suggested implementation for triggering the method only when needed

Comment (bulgakovk) [reply wallets/exodus-extension/src/extension/client.ts]: Sounds good, thanks!

**Pull request merged**
